Append node

smbea opened the pull request:
Related to #5

barmac commented on assets/append.css:
```diff
@@ -0,0 +1,44 @@
+    transition: 0.1s;
+  }
+
+  & .djs-append-node-circle:hover,
+  & :hover .djs-append-node-circle:hover {
```
Aren't these two equal?

barmac commented on test/spec/features/append/AppendNodeProvider.spec.js:
```diff
@@ -0,0 +1,116 @@
+        const allowed = appendNodeProvider.isAllowed(shape);
+
+        // then
+        expect(allowed).to.be.false;
```
Note that if this ever breaks, we will only see `expected true to be false`. Perhaps it could be a more future-proof solution to place `it` inside of iteration, so that the test name takes form of "it should disallow append node for {elementId}"

barmac commented on test/spec/features/append/AppendNodeProvider.spec.js:
```diff
@@ -0,0 +1,116 @@
+
+describe('<AppendNodeProvider>', function() {
+
+  function bootstrap(diagramXML) {
+
+    return function() {
+      return bootstrapModeler(diagramXML, {
+        additionalModules: [
+          Append,
+          CreateAppendAnythingModule
+        ]
+      })();
+    };
+  }
+
+  beforeEach(bootstrap(diagramXML));
```
Why not this:

```javascript
beforeEach(bootstrapModeler(diagramXML, {
  additionalModules: [
    Append,
    CreateAppendAnythingModule
  ]
}));
```

barmac commented on lib/features/append/AppendNode.js:
```diff
@@ -0,0 +1,294 @@
+
+  const node = overlays.get(this._overlayId);
+
+  // notidfy others
```
Typo:
```suggestion
  // notify others
```

barmac replied:
Also, do we need this?

barmac commented:
I am worried that there is too much motion in this feature. We don't support `prefers-reduced-motion: reduce`:

https://github.com/camunda/improved-canvas/assets/28307541/efef526d-5be7-4371-9662-0c8cfee4498e

At the same time, I had a problem discovering it in the first place 🤡

smbea commented:
I simplified/improved some things according to your suggestions.

Regarding "notifying others": I believe we should keep the `trigger` which will be helpful for tracking and such, if we decide to. On `create`: I added this because we also have it for the context pad/palette, etc components. But I just learned we use it in [TouchInteractionEvents](https://github.com/bpmn-io/diagram-js/blob/develop/lib/features/touch/TouchInteractionEvents.js). I'm not sure if this can be extended since the functions aren't accessible.

> I am worried that there is too much motion in this feature. We don't support prefers-reduced-motion: reduce:

I can write this in the slack channel and ask for opinions, including Leah's.

smbea commented:
@barmac I simplified the UI as we talked about

barmac commented:
There is one thing which makes me wonder. Why do we attach the append node as the root element overlay instead of element overlay?

smbea commented:
> There is one thing which makes me wonder. Why do we attach the append node as the root element overlay instead of element overlay?

We don't actually @barmac. I updated to attach to element.
